src/main/java/hng_java_boilerplate/product/controller/ProductController.java

```diff
@@ -61,4 +62,44 @@ public ResponseEntity<ProductSearchDTO> searchProducts(
         productSearchDTO.setSuccess(true);
         return new ResponseEntity<>(productSearchDTO, HttpStatus.OK);
     }
+
+
+    @GetMapping("/{productId}/stock")
+    public ResponseEntity<ProductStockDTO> getProductStock(@PathVariable String productId) {
+        Product product = productService.getProductStock(productId);
+        ProductStockDTO productStockDTO = new ProductStockDTO(
+                product.getId(),
+                product.getCurrentStock(),
+                product.getLastUpdated()
+        );
+        return new ResponseEntity<>(productStockDTO, HttpStatus.OK);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_USER') or hasRole('ROLE_ADMIN')")
```

Comment:
If any authenticated user should be able to create a product, is the preAuthorize needed?

Reply:
If only the admin should be able to create a product, remove the role_user, but if a user should be able to create product and is not limited to admin, remove the preauthorize, it's not needed